docusaurus/docs/cms/admin-panel-customization/homepage.md

```diff
@@ -0,0 +1,761 @@
+      },
+      id: 'my-custom-widget',
+      pluginId: pluginId,
+    });
+  },
+  
+  bootstrap() {},
+  // ...
+};
+```
+
+</TabItem>
+</Tabs>
+
+#### Widget API reference
+
+The `app.widgets.register()` method can take either a single widget configuration object or an array of configuration objects. Each widget configuration object can accept the following properties:
+
+| Property    | Type                   | Description                                           | Required |
+|-------------|------------------------|-------------------------------------------------------|----------|
+| `icon`      | `React.ComponentType`  | Icon component to display beside the widget title     | Yes      |
+| `title`     | `MessageDescriptor`    | Title for the widget with translation support         | Yes      |
+| `component` | `() => Promise<React.ComponentType>` | Async function that returns the widget component | Yes      |
+| `id`        | `string`               | Unique identifier for the widget                      | Yes      |
+| `link`      | `Object`               | Optional link to add to the widget (see link object properties)| No       |
+| `pluginId`  | `string`               | ID of the plugin registering the widget               | No       |
+| `permissions` | `Permission[]`       | Permissions required to view the widget               | No       |
+
+**Link object properties:**
+
+If you want to add a link to your widget (e.g., to navigate to a detailed view), you can provide a `link` object with the following properties:
+
+| Property | Type                | Description                                    | Required |
+|----------|---------------------|------------------------------------------------|----------|
+| `label`  | `MessageDescriptor` | The text to display for the link               | Yes      |
+| `href`   | `string`            | The URL where the link should navigate to      | Yes      |
+
+### Creating a widget component
+
+Widget components should be designed to display content in a compact and informative way. 
+
+Here's how to implement a basic widget component:
+
+<Tabs groupId="js-ts">
+<TabItem value="javascript" label="JavaScript">
+
+```jsx title="src/plugins/my-plugin/admin/src/components/MyWidget/index.js"
+import React, { useState, useEffect } from 'react';
+import { Widget } from '@strapi/admin/strapi-admin';
+
+const MyWidget = () => {
+  const [loading, setLoading] = useState(true);
+  const [data, setData] = useState(null);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    // Fetch your data here
+    const fetchData = async () => {
+      try {
+        // Replace with your actual API call
+        const response = await fetch('/my-plugin/data');
+        const result = await response.json();
+        
+        setData(result);
+        setLoading(false);
+      } catch (err) {
+        setError(err);
+        setLoading(false);
+      }
+    };
+
+    fetchData();
+  }, []);
```

Comment:
This is more about the larger docs strategy in general, but do we need to go into so much detail about this? Also because fetching in useEffect directly is not recommended, [there are many tools](https://react.dev/learn/build-a-react-app-from-scratch#data-fetching) to handle this

Reply:
Thanks for the feedback. I'd say we need this level of details, yes. Users are asking for it.
Regarding the use of fetch, to speed up the improvement process (we're on release day), could you please directly provide which code you would recommend using? 🙏

Reply:
if we do need to show detailed code snippets then let's stick to this, as other solutions require installing a library, which would add friction

Reply:
If you wanna mention outside of the snippet that it's recommended to use a data fetching library instead of doing it manually though, I would name drop [TanStack
Query](https://tanstack.com/query/v3/)